Automattic/wp-calypso
# Signup: New flow to kick off an import post site creation (behind dev flag)

This PR adds a new flow called `from-site` with a new first step `import-from-url`.
The new step is intended to collect the URL from which the customer wants to import.

This is meant as a starting point for adding this flow. It's behind a flag so we can confidently land it and add-on in smaller increments in follow-up PRs.

### To Test

* ~~Needs D17199-code & a patched API sandbox for now~~ Sandboxing no longer required.
* Run this branch locally
* Visit http://calypso.localhost:3000/start/from-site
* It should redirect to http://calypso.localhost:3000/start/from-site/import-from-url & you should see a new step with a form to enter your URL.
* Submitting it should progress to the next flow step regardless of the field content (for now)
* Entering text in the field should submit the URL to the `is-site-importable` endpoint
* Visiting from http://calypso.localhost:3000/start/from-site/import-from-url?https://example.com should pre-populate the input field & submit the URL to the `is-site-importable` endpoint
* While the network request completes, the form should be disabled & it should be re-enabled once the request completes
* Completing the flow should create a site with the Radcliffe 2 theme

### Next Steps

Some things that will make this actually useful:
* Reduxify importer state ( #26512  ) 
* Modularize reusable components
* Appropriately handle the server-side validation / importability check
* Investigate automatically choosing a site address
* Land at the import section & kick off an import
